service/src/main/java/org/cbioportal/genome_nexus/service/internal/GenomicLocationAnnotationServiceImpl.java

```diff
@@ -88,7 +88,7 @@ public GenomicLocationAnnotationServiceImpl(CachedVariantRegionAnnotationFetcher
     public VariantAnnotation getAnnotation(GenomicLocation genomicLocation)
         throws VariantAnnotationNotFoundException, VariantAnnotationWebServiceException
     {
-        VariantAnnotation variantAnnotation = this.variantAnnotationService.getAnnotation(this.notationConverter.genomicToEnsemblRestRegion(genomicLocation));
+        VariantAnnotation variantAnnotation = this.variantAnnotationService.getAnnotation(this.genomicLocationToVariantFormat.convert(genomicLocation));
```

Comment:
Yes, this is good. The concrete instance of the functional interface for genomicLocationToVariantFormat is set according to isRegionAnnotationEnabled(). When we have an attached vep (from genome-nexus-vep) we send annotation requests in RestRegion format to VEP - so genomicLocationToVariantFormat.convert() will be set to notationConverter::genomicToEnsemblRestRegion. Otherwise, we will be sending requests (in Hgvs format) to the public Ensembl VEP API - so so genomicLocationToVariantFormat.convert() will be set to notationConverter:: genomicToHgvs.
This change fixes a bug - where we were always converting the genomicLocation to RestRegion no matter what configuration we were in. However ... seeing this bug now, I don't see how any request could succeed when sent to the POST endpoint with a genomicLocation object supplied by the user. It seems like those should also be improperly converted to RestRegion and then fail annotation in the Ensembl Rest API (which expect hgvs). But this request succeeds on the current public genome nexus:
curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' -d '[ \ 
   { \ 
     "chromosome": "7", \ 
     "start": 55242468, \ 
     "end": 55242469, \ 
     "referenceAllele": "AT", \ 
     "variantAllele": "GC" \ 
   } \ 
 ]' 'https://www.genomenexus.org/annotation/genomic'
Shouldn't this query also (with the current code) be converted to Ensembl Rest Region and then be sent to the Public Ensembl API and fail? I guess this endpoint actually is supplying a *list* of GenomicLocations, so the call is to getAnnotations(List<GenomicLocation>, but that function (below) also has the bug. AHA .. it calls the function with the extra arguments (genomicLocations, isoformOverrideSource, token, and fields) and that function does not have the bug. Ok, it makes sense now.